src/main/java/org/example/services/BucketService.java

```
@@ -0,0 +1,130 @@
+import java.io.InputStreamReader;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BucketService {
+    FileImportDao fileImportDao;
+    DatabaseConnector databaseConnector;
+    FileImportValidator fileImportValidator;
+
+    public BucketService(final FileImportDao fileImportDao, final FileImportValidator fileImportValidator) {
+        this.fileImportDao = fileImportDao;
+        this.fileImportValidator = fileImportValidator;
+    }
+
+
+    public int importJobRoles() {
+        System.out.println("Test");
+        int result = 0;
+
+        AWSCredentials credentials = new BasicAWSCredentials(
+                System.getenv().get("AWS_ACCESS_KEY_ID"),
+                System.getenv().get("AWS_SECRET_ACCESS_KEY")
+
+        );
+        AmazonS3 s3client = AmazonS3ClientBuilder
+                .standard()
+                .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                .withRegion(Regions.EU_WEST_1)
+                .build();
+
+        String folderName = "the_extra_semicolon/imports/";
+        String filename;
+        String fullFileName;
+        ListObjectsV2Request req = new ListObjectsV2Request().withBucketName(System.getenv().get("S3_BUCKET")).withPrefix(folderName);
+        ListObjectsV2Result listResult;
+        listResult = s3client.listObjectsV2(req);
+        for (S3ObjectSummary objectSummary : listResult.getObjectSummaries()) {
+            System.out.println(" - " + objectSummary.getKey() + "  " + "(size = " + objectSummary.getSize() + ")");
+            filename = objectSummary.getKey();
+           // String[] splitFilepath = objectSummary.getKey().split("/");
+           // filename = splitFilepath[2];
+            System.out.println(filename);
+            //filename = objectSummary.getKey();
+
+            System.out.format("Downloading %s from S3 bucket %s...\n",
+                    filename, System.getenv().get("S3_BUCKET"));
+            try {
```

This code should be refactored into a new function.

Lines 74-93 where the function takes a filename and returns a list of rows to process.

Leave within the try catch and then let it throw exceptions which the importJobRoles can handle